### PR TITLE
Implement cron

### LINKED ISSUE #24: Implement simple cron
On weave v0.19.0, cron tasks have been implemented. Write boilerplate code for cron and implement a simple cron task.

### PR DESCRIPTION
**Pull request by orkunkl**
This PR implements basic `cron` boilerplate. `DeleteTimedStateMsg` is executed at `TimedState.DeleteAt`

Resolved #24

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -88,6 +90,29 @@ func Stack(issuer weave.Address, minFee coin.Coin) weave.Handler {
 	return Chain(authFn, minFee).WithHandler(Router(authFn, issuer))
 }
 
+// CronStack wires up a standard router with a cron specific decorator chain.
+// This can be passed into BaseApp.
+// Cron stack configuration is a subset of the main stack. It is using the same
```
Comment (husio) [inline app/app.go:95]: Cron stack does not have to be a subset of the main stack. It all depends on what messages the cron should have access to.
Cron messages can be handled by different set of handlers and different middleware configuration. That allows you to for example not charge any fee for their execution or disable/enable certain messages and therefore functionality.


This is not a wrong comment. I want to make sure that you understand why we did it like this in bnsd.

File: app/crontask.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+func (taskMarshaler) MarshalTask(auth []weave.Condition, msg weave.Msg) ([]byte, error) {
+	t := CronTask{
+		Authenticators: auth,
+	}
+
+	switch msg := msg.(type) {
+	default:
+		return nil, errors.Wrapf(errors.ErrType, "unsupported message type: %T", msg)
+
+		case *custom.DeleteTimedStateMsg:
```
Comment (husio) [inline app/crontask.go]: Please run `go fmt` on this file.

**Review by husio — APPROVED** [review]

File: app/crontask.go (hunk 2 above)
Comment (orkunkl) [reply app/crontask.go]: Could not make vscode auto fmt work. Fixed the issue with using a githook.

File: app/app.go (hunk 1 above)
Comment (orkunkl) [reply app/app.go:95]: I see :+1: I will explain this in https://github.com/iov-one/docs/issues/50

**Pull request merged**